client/issue-lib.js

```diff
@@ -0,0 +1,146 @@
+(function() {
+'use strict';
+
+if (window.Issue) {
+  return;
+}
+
+// Colours are copied from:
+// https://www.google.com/design/spec/style/color.html#color-color-palette
+var _reviewLevelMetadata = {
+  'daily': {
+    query: {label: 'Update-Daily'},
+    outOfSLOColor: '#B71C1C',  // Red 900
+  },
+  'weekly': {
+    query: {label: 'Update-Weekly'},
+    outOfSLOColor: '#C62828',  // Red 800
+  },
+  'fortnightly': {
+    query: {label: 'Update-Fortnightly'},
+    outOfSLOColor: '#D32F2F',  // Red 700
+  },
+  'monthly': {
+    query: {label: 'Update-Monthly'},
+    outOfSLOColor: '#E53935',  // Red 600
+  },
+  'quarterly': {
+    query: {label: 'Update-Quarterly'},
+    outOfSLOColor: '#F44336',  // Red 500
+  },
+  'none (P0)': {
+    query: {label: 'Pri-0', '-has': 'update'},
+  },
+  'none (P1)': {
+    query: {label: 'Pri-1', '-has': 'update'},
+  },
+  'none (P2)': {
+    query: {label: 'Pri-2', '-has': 'update'},
+  },
+  'none (P3)': {
+    query: {label: 'Pri-3', '-has': 'update'},
+  },
+};
+var _defaultReviewLevel = 'none';
+var _inSLOColor = '#4CAF50';  // Green 500
+var _noSLOColor = '#757575';  // Grey 600
+
+// This class uses C style programming with JSON objects (representing issues)
+// and static methods.
+// This is to maintain the ability to clone the object in chromez-behaviours.js.
```

Comment:
You don't need to do this to allow objects to be cloned. You can either explicitly provide a clone() method, or serialize the representation automatically with json. Either way, methods on the object won't get in the way.

As I mentioned in the other comments, *this is a major detail that needs to be included in design documentation*. We should have been able to review this proposal before you went to the effort of coding it up.